src/common/github/pull-request-test-workflow.ts

```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
       steps: [{uses: 'ottofeller/github-actions/npm-run@main', with: {'node-version': 16, command, directory}}],
     })
 
+    const customJob = (steps: JobStep[]): github.workflows.Job => ({
```

Comment:
There is a `job` function in `src/common/github/job.ts`. We can utilise that one here adding a second optional argument.
```typescript
export const job = (steps: Array<JobStep>, runsOn = ['ubuntu-latest']): Job => ({
  runsOn,
  permissions: {contents: projen.github.workflows.JobPermission.READ},
  steps,
})
```

However I'm not sure we pass `runsOn` anywhere - moreover seems like it's not even possible with current setup. So you'd be able to use the function as is.